crates/refunder/src/refund_service.rs

```diff
@@ -143,4 +172,143 @@ impl RefundService {
+                        );
+                        Ok(None)
+                    }
+                }
+            }
+        });
+        let encoded_ethflow_orders: Vec<Result<Option<EncodedEthflowOrder>>> =
+            stream::iter(futures)
+                .buffer_unordered(10)
+                .collect::<Vec<Result<Option<EncodedEthflowOrder>>>>()
+                .await;
+        let encoded_ethflow_orders: Vec<EncodedEthflowOrder> = encoded_ethflow_orders
+            .into_iter()
+            .flatten()
+            .flatten()
+            .collect();
+
+        self.submitter.submit(uids, encoded_ethflow_orders).await?;
+        Ok(())
+    }
+}
+
+fn order_to_ethflow_data(
+    order: Order,
+    ethflow_order_placement: EthOrderPlacement,
+) -> EncodedEthflowOrder {
+    (
+        H160(order.buy_token.0),
+        H160(order.receiver.unwrap().0), // ethflow orders have always a
+        // receiver. It's enforced by the contract.
+        big_decimal_to_u256(&order.sell_amount).unwrap(),
+        big_decimal_to_u256(&order.buy_amount).unwrap(),
+        Bytes(order.app_data.0),
+        big_decimal_to_u256(&order.fee_amount).unwrap(),
+        ethflow_order_placement.valid_to as u32, // Unwrap can never fail, as the value
+        // is not None for ethflow orders
+        false, // ethflow orders are always fill or kill orders
+        0i64,  // quoteId is not important for refunding and will be ignored
+    )
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use database::byte_array::ByteArray;
+    use number_conversions::u256_to_big_decimal;
+
+    #[test]
+    fn test_order_to_ethflow_data() {
+        let buy_token = ByteArray([1u8; 20]);
+        let receiver = ByteArray([3u8; 20]);
+        let sell_amount = U256::from_dec_str("1").unwrap();
+        let buy_amount = U256::from_dec_str("2").unwrap();
+        let app_data = ByteArray([3u8; 32]);
+        let fee_amount = U256::from_dec_str("3").unwrap();
+        let valid_to = 234u32;
+
+        let order = Order {
+            buy_token,
+            receiver: Some(receiver),
+            sell_amount: u256_to_big_decimal(&sell_amount),
+            buy_amount: u256_to_big_decimal(&buy_amount),
+            valid_to: valid_to.into(),
+            app_data,
+            fee_amount: u256_to_big_decimal(&fee_amount),
+            ..Default::default()
+        };
+        let ethflow_order = EthOrderPlacement {
+            valid_to: valid_to.into(),
+            ..Default::default()
+        };
+        let expected_encoded_order = (
+            H160(order.buy_token.0),
+            H160(order.receiver.unwrap().0),
+            big_decimal_to_u256(&order.sell_amount).unwrap(),
+            big_decimal_to_u256(&order.buy_amount).unwrap(),
+            Bytes(order.app_data.0),
+            big_decimal_to_u256(&order.fee_amount).unwrap(),
+            ethflow_order.valid_to as u32,
+            false,
+            0i64,
+        );
```

Comment:
Not sure how useful this test is because it mostly copies the implementation it tries to test. 🤷‍♂️

Reply:
Yes, it only shows that an unwrap is not panicking. Not sure how useful it is. But I think it helps a tiny tiny bit ...